src/CodeGenerator/Generator/RequestSerializer/RestXmlSerializer.php

```diff
@@ -16,9 +17,28 @@
  *
  * @internal
  */
-class XmlDumper
+class RestXmlSerializer implements Serializer
 {
-    public function dumpXml(StructureMember $member, string $payloadProperty): string
+    public function getContentType(): string
+    {
+        return 'application/xml';
+    }
+
+    public function generateForShape(Operation $operation, StructureShape $shape): string
+    {
+        $httpMethod = $operation->getHttpMethod();
+        if (\in_array($httpMethod, ['DELETE', 'GET', 'HEAD'])) {
+            return 'return "";';
+        }
+
+        if (null === $shape->getPayload()) {
+            return 'return "";';
+        }
+
+        throw new \RuntimeException(sprintf('Generating an XML a body for "%s %s::%s" is not supported.', $httpMethod, $operation->getService()->getName(), $operation->getName()));
```

Comment:
Im not 100% sure this is correct. But we will get some exceptions if we try to generate an API operation that are using this.